hlapipeline/alignimages.py

```diff
@@ -3,65 +3,173 @@
+            fitsfilename = fitsfilename[0]
+
+            if not os.path.exists(fitsfilename):
+                imghdu = fits.open(fitsfilename)
+                imgprimaryheader = imghdu[0].header
+                totalInputList += aqutils.retrieve_observation(imgprimaryheader['ASN_ID'].strip())
+            else: totalInputList.append(fitsfilename)
+    print("TOTAL INPUT LIST: ",totalInputList)
+    # TODO: add trap to deal with non-existent (incorrect) rootnames
+    # TODO: add "Archive" option to aqutils.retrieve_observation
+    # TODO: Address issue about how the code will retrieve association information if there isn't a local file to get 'ASN_ID' header info
+    return(totalInputList)
+
+
+def perform_align(input_list):
     """Main calling function.
 
     Parameters
     ----------
     input_list : list
-        List of one or more input image(s) and/or association file(s) to align.
+        List of one or more IPPSSOOTs (rootnames) to align.
 
     Returns
     -------
     Nothing for now.
 
     """
 
+    # Define astrometric catalog list in priority order
+    catalogList = ['GAIADR2', 'GSC241']
+    numCatalogs = len(catalogList)
+
     # 1: Interpret input data and optional parameters
+    print("-------------------- STEP 1: Get data --------------------")
+    imglist = check_and_get_data(input_list)
+    print("\nSUCCESS")
 
     # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
-    imglist = input_list
+    print("-------------------- STEP 2: Filter data --------------------")
+    filteredTable = filter.analyze_data(imglist)
+
+    # Check the table to determine if there is any viable data to be aligned.  The
+    # 'doProcess' column (bool) indicates the image/file should or should not be used
+    # for alignment purposes.
+    if filteredTable['doProcess'].sum() == 0:
+        print("No viable images in filtered table - no processing done.\n")
+        return
+
+    # Get the list of all "good" files to use for the alignment
+    processList = filteredTable['imageName'][np.where(filteredTable['doProcess'])]
+    processList = list(processList) #Convert processList from numpy list to regular python list
+    print("\nSUCCESS")
 
     # 3: Build WCS for full set of input observations
-    refwcs = amutils.build_reference_wcs(imglist)
+    print("-------------------- STEP 3: Build WCS --------------------")
+    refwcs = amutils.build_reference_wcs(processList)
+    print("\nSUCCESS")
 
     # 4: Retrieve list of astrometric sources from database
-    reference_catalog = generate_astrometric_catalog(imglist, catalog='GAIADR2')
-
+    print("-------------------- STEP 4: Detect astrometric sources --------------------")
+    # While loop to accommodate using multiple catalogs
+    doneFitting = False
+    catalogIndex = 0
+    while not doneFitting:
+        reference_catalog = generate_astrometric_catalog(processList, catalog='GAIADR2', existing_wcs=refwcs)
```

Comment:
The inclusion of the 'existing_wcs' parameter here does not make sense, since the reference catalog created in this call should only return positions in sky coordinates, not positions transformed to ANY tangent plane.

Reply:
Issue resolved by commit [7eff0bb](https://github.com/spacetelescope/hlapipeline/commit/7eff0bb4c4d7ceb526f1580bccf83985e9c415b5)